frame/beefy/src/tests.rs

```diff
@@ -140,7 +157,625 @@ fn validator_set_updates_work() {
+	// same payload signed by the same key
+	let equivocation_proof = generate_equivocation_proof(
+		set_id,
+		BeefyKeyring::Alice.public(),
+		(1, payload1.clone(), &BeefyKeyring::Bob),
+		(1, payload1.clone(), &BeefyKeyring::Bob),
+	);
+	// expect invalid equivocation proof
+	assert!(!check_equivocation_proof::<_, _, Keccak256>(equivocation_proof));
+
+	// generate an equivocation proof, with two votes in different rounds for
+	// different payloads signed by the same key
+	let equivocation_proof = generate_equivocation_proof(
+		set_id,
+		BeefyKeyring::Alice.public(),
+		(1, payload1.clone(), &BeefyKeyring::Bob),
+		(2, payload2.clone(), &BeefyKeyring::Bob),
+	);
+	// expect invalid equivocation proof
+	assert!(!check_equivocation_proof::<_, _, Keccak256>(equivocation_proof));
+
+	// generate an equivocation proof, with two votes in the same round for
+	// different payloads signed by the same key
+	let payload2 = Payload::from_single_entry(MMR_ROOT_ID, vec![128]);
+	let equivocation_proof = generate_equivocation_proof(
+		set_id,
+		BeefyKeyring::Alice.public(),
+		(1, payload1, &BeefyKeyring::Bob),
+		(1, payload2, &BeefyKeyring::Bob),
+	);
+	// expect valid equivocation proof
+	assert!(check_equivocation_proof::<_, _, Keccak256>(equivocation_proof));
+}
+
+#[test]
+fn report_equivocation_current_set_works() {
+	let authorities = test_authorities();
+
+	new_test_ext_raw_authorities(authorities).execute_with(|| {
+		assert_eq!(Staking::current_era(), Some(0));
+		assert_eq!(Session::current_index(), 0);
+
+		start_era(1);
+
+		let block_num = System::block_number();
+		let validator_set = Beefy::validator_set().unwrap();
+		let authorities = validator_set.validators();
+		let set_id = validator_set.id();
+		let validators = Session::validators();
+
+		// make sure that all validators have the same balance
+		for validator in &validators {
+			assert_eq!(Balances::total_balance(validator), 10_000_000);
+			assert_eq!(Staking::slashable_balance_of(validator), 10_000);
+
+			assert_eq!(
+				Staking::eras_stakers(1, validator),
+				pallet_staking::Exposure { total: 10_000, own: 10_000, others: vec![] },
+			);
+		}
+
+		assert_eq!(authorities.len(), 2);
+		let reporter_index = 0;
+		let _reporter_key = authorities[reporter_index].clone();
+		let equivocation_authority_index = 1;
+		let equivocation_key = &authorities[equivocation_authority_index];
+		let equivocation_keyring = BeefyKeyring::from_public(equivocation_key).unwrap();
+
+		let payload1 = Payload::from_single_entry(MMR_ROOT_ID, vec![42]);
+		let payload2 = Payload::from_single_entry(MMR_ROOT_ID, vec![128]);
+		// generate an equivocation proof, with two votes in the same round for
+		// different payloads signed by the same key
+		let equivocation_proof = generate_equivocation_proof(
+			set_id,
+			// FIXME: test only passes for equivocation auto-report, why?
+			// _reporter_key.clone(),
+			equivocation_key.clone(),
```

Comment:
@andresilva I can't figure out why slashing doesn't happen when equivocation is reported by any other validator in the active set. Any ideas?

Reply:
Nevermind, I got it. I was using it wrong. Will fix it shortly.